src/VisualStudio/Next/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -43,6 +49,17 @@ internal partial class ServiceHubRemoteHostClient : RemoteHostClient
             }
         }
 
+        private static void RegisterWorkspaceHost(Workspace workspace)
+        {
+            var vsWorkspace = workspace as VisualStudioWorkspaceImpl;
+            if (vsWorkspace == null)
+            {
+                return;
+            }
+
+            vsWorkspace.ProjectTracker.RegisterWorkspaceHost(new WorkspaceHost(vsWorkspace));
```

Comment:
I see you didn't move UpdateSolutionProperties so I guess this depends on the fact VSWorkspaceHost being the very first workspacehost?